src/app/client/src/app/modules/dashboard/components/course-progress/course-progress.component.ts

```diff
@@ -419,6 +432,47 @@ export class CourseProgressComponent implements OnInit, OnDestroy, AfterViewInit
         this.populateCourseDashboardData();
       });
   }
+  getReportUpdatedOnDate() {
+    const batchId = _.get(this.queryParams, 'batchIdentifier');
+    const reportParams = {
+      'course-progress-reports': `course-progress-reports/report-${batchId}.csv`,
```

Comment:
All these API logic should go into service rather than a component. Why are we not using course-progress service?

Reply:
Fixed